src/lib/tabs/tab-header.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  ViewChild,
+  Component,
+  Input,
+  NgZone,
+  QueryList,
+  ElementRef,
+  Renderer, ViewEncapsulation, ContentChildren, Output, EventEmitter
+} from '@angular/core';
+import {RIGHT_ARROW, LEFT_ARROW, ENTER, Dir, LayoutDirection} from '../core';
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+import 'rxjs/add/operator/map';
+
+export type ScrollDirection = 'after' | 'before';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-tab-header',
+  templateUrl: 'tab-header.html',
+  styleUrls: ['tab-header.css'],
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    '[class.md-tab-header]': 'true',
+    '[class.scroll]': '_isScrollingEnabled()',
```

Comment:
I agree that this is not ideal. Do you have any suggestions on how to reduce the amount of times we call this but still have an up to date pagination? It seems that checking this in ngAfterViewChecked will be a little better but still way more than we should need.